### PR TITLE
sys-apps/lshw: fix musl build, update EAPI

### PR DESCRIPTION
**Pull request by hanetzer**
Package-Manager: Portage-2.3.8, Repoman-2.3.3
Signed-off-by: Marty E. Plummer <hanetzer@protonmail.com>

### CODE REVIEW

File: sys-apps/lshw/files/lshw-02.18b-musl.patch [hunk 1]
```diff
@@ -0,0 +1,58 @@
+diff -aurP a/src/core/dasd.cc b/src/core/dasd.cc
```
Comment (Whissi) [inline sys-apps/lshw/files/lshw-02.18b-musl.patch]: Please tell us more about the patch origin.

Comment (hanetzer) [reply sys-apps/lshw/files/lshw-02.18b-musl.patch]: https://github.com/lyonel/lshw/commit/cd690bff1516b40fecd5ec4a7f6619e5bffc3cf0
https://patchwork.ozlabs.org/patch/655854/
^ Based on some changes made in these patches, adjusted.

Comment (a17r) [reply sys-apps/lshw/files/lshw-02.18b-musl.patch]: @hanetzer Patch origin and reason should be explained in the patch header

Comment (Whissi) [reply sys-apps/lshw/files/lshw-02.18b-musl.patch]: Sorry, I have to reject this patch. Please see @vapier comment in https://github.com/lyonel/lshw/pull/19

Comment (SoapGentoo) [reply sys-apps/lshw/files/lshw-02.18b-musl.patch]: just copy the `c_str()` using `strcpy` into a temporary, muck with that, and dont forget to deallocate it afterwards

**Comment by monsieurp** [comment]
Closing as per @Whissi 's comment.

**Pull request closed without merging**